### ISSUE TITLE
Bug: After the latest update to v1.1.0 this plug took down my whole Homebridge service

### ISSUE THREAD

**Issue by ydogandjiev** [OpenWonderLabs/homebridge-switchbot#99]
### Describe The Bug

After the latest update to v1.1.0 this plug took down my whole Homebridge service

### To Reproduce

Install v1.1.0, restart the Homebridge service

### Expected behavior

The plugin should not take down my service

### Relevant log output

```shell
These were some of the relevant errors I saw in the logs:
[11/17/2021, 6:49:25 PM] [SwitchBot] You must set your Bot to Press or Switch Mode
[11/17/2021, 6:49:25 PM] [SwitchBot] Adding new accessory: SwitchBot 1 Bot DeviceID: E21C43AC2D24
[11/17/2021, 6:49:25 PM] [SwitchBot] You must set your Bot to Press or Switch Mode
[11/17/2021, 6:49:26 PM] TypeError: Cannot read properties of undefined (reading 'deviceType')
    at new Bot (/usr/lib/node_modules/@switchbot/homebridge-switchbot/src/devices/bots.ts:68:25)
    at SwitchBotPlatform.createBot (/usr/lib/node_modules/@switchbot/homebridge-switchbot/src/platform.ts:464:7)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
ERROR: uncaughtException: Cannot read properties of undefined (reading 'deviceType')
TypeError: Cannot read properties of undefined (reading 'deviceType')
    at new Bot (/usr/lib/node_modules/@switchbot/homebridge-switchbot/src/devices/bots.ts:68:25)
    at SwitchBotPlatform.createBot (/usr/lib/node_modules/@switchbot/homebridge-switchbot/src/platform.ts:464:7)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
[11/17/2021, 6:49:26 PM] TypeError: Cannot read properties of undefined (reading 'deviceType')
    at new Bot (/usr/lib/node_modules/@switchbot/homebridge-switchbot/src/devices/bots.ts:68:25)
    at SwitchBotPlatform.createBot (/usr/lib/node_modules/@switchbot/homebridge-switchbot/src/platform.ts:464:7)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
ERROR: uncaughtException: Cannot read properties of undefined (reading 'deviceType')
TypeError: Cannot read properties of undefined (reading 'deviceType')
    at new Bot (/usr/lib/node_modules/@switchbot/homebridge-switchbot/src/devices/bots.ts:68:25)
    at SwitchBotPlatform.createBot (/usr/lib/node_modules/@switchbot/homebridge-switchbot/src/platform.ts:464:7)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

### Config for homebridge-switchbot

```shell
{
            "name": "SwitchBot",
            "credentials": {
                "openToken": "<hidden>",
                "notice": "Keep your token a secret!"
            },
            "options": {
                "hide_device": [
                    "E21C43AC2D24",
                    "DE27B8505B22",
                    "C7D35B609D5D"
                ]
            },
            "devicediscovery": false,
            "platform": "SwitchBot"
        },
```

### Device & Model

I have multiple devices

### Node.js Version

v16.13.0

### NPM Version

v8.1.0

### Homebridge Version

v1.3.6

### Homebridge Switchbot Plugin Version

v1.1.0

### Homebridge Config UI X Plugin Version

v4.41.2

### Operating System

Raspbian

**Comment by lgranger** [comment]
I also had my homebridge server crash when I upgraded to v1.1.0, but I'm not sure that they are causally linked, since I downgraded back through the versions to v 1.0.0 and was still unable to restore functionality. 

At first I was getting the same `TypeError: Cannot read properties of undefined (reading 'deviceType')` error as @ydogandjiev, but after opening the settings for the switchbot plugin and then restarting that error went away and my whole homebridge server isn't crash looping anymore.

Now it is not loading the switchbot devices properly. Logs say that it is loading them from cache after it fails to load them due to missing Device Type:

```[11/17/2021, 7:18:12 PM] [SwitchBot] Initializing SwitchBot platform...
[11/17/2021, 7:18:12 PM] [SwitchBot] Finished initializing platform: SwitchBot
[11/17/2021, 7:18:12 PM] [SwitchBot] "The devices config section is missing the \"Device Type\" in the config, Check Your Conifg."
[11/17/2021, 7:18:12 PM] [SwitchBot] {}
......
[11/17/2021, 7:28:00 PM] [SwitchBot] Loading accessory from cache: Heater
[11/17/2021, 7:28:00 PM] [SwitchBot] Loading accessory from cache: Vacuum On
[11/17/2021, 7:28:00 PM] [SwitchBot] Loading accessory from cache: Vacuum Home
[11/17/2021, 7:28:00 PM] [SwitchBot] Loading accessory from cache: Bedroom 
[11/17/2021, 7:28:00 PM] [SwitchBot] Loading accessory from cache: Office 
[11/17/2021, 7:28:00 PM] [SwitchBot] Loading accessory from cache: Outside 
[11/17/2021, 7:28:00 PM] [SwitchBot] Loading accessory from cache: Curtain
[11/17/2021, 7:28:00 PM] [SwitchBot] Loading accessory from cache: Living Room
[11/17/2021, 7:28:00 PM] [SwitchBot] Loading accessory from cache: Kitchen
```

I see the devices, but the temp/humidity sensors are returning 0 for all values (including battery level), and I can't control any of the switches.

**Comment by donavanbecker** [comment]
@ydogandjiev Your hide_device section is no longer needed, remove it and remove and re-add

@lgranger i need to see your config.

**Comment by ydogandjiev** [comment]
> @ydogandjiev Your hide_device section is no longer needed, remove it and remove and re-add

@donavanbecker, thanks for the quick reply but this didn't work. I removed the SwitchBot plugin completely (including the config) and re-installed it just adding my token in the config. Now at least Homebridge starts up but none of my SwitchBot devices show up. I see this error in the logs:
```
[11/17/2021, 10:54:32 PM] [SwitchBot] Initializing SwitchBot platform...
[11/17/2021, 10:54:32 PM] [SwitchBot] "The devices config section is missing the \"Device Type\" in the config, Check Your Conifg."
```

Seems like this latest version still has some serious bugs in it. I am rolling back to v1.0.2 to get myself unblocked. Will not be upgrading again until this issue is resolved.

**Comment by ydogandjiev** [comment]
Hmm, rolling back to v1.0.2 didn't fix the issue this time around. :( I had to delete the two SwitchBots from my account that appeared to be causing the issue. Thankfully I am not using them at the moment. Hopefully whatever caused this regression gets resolved soon.

**Comment by donavanbecker** [comment]
@ydogandjiev, can you try Beta: 

[![npm version](https://badgen.net/badge/homebridge-switchbot@beta/>=1.1.1-beta.8/orange)](https://www.npmjs.com/package/@switchbot/homebridge-switchbot)

[Click here to find out how to install Beta](https://github.com/OpenWonderLabs/homebridge-switchbot/wiki/Beta-Version)

**Comment by krukje** [comment]
I'm having the same issue, also after installing 1.1.1b8. Log shows:
```
[18/11/2021, 14:30:39] [SwitchBot] The devices config section is missing the *Device ID* in the config, Check Your Conifg.
[18/11/2021, 14:30:39] [SwitchBot] The devices config section is missing the *Device Type* in the config, Check Your Conifg.
[18/11/2021, 14:30:39] [SwitchBot] You must set your Bot to Press or Switch Mode
```

I've removed the switchbot (only keeping curtains and airco control), removed the plugin & reinstalled but still get the above error in 1.1.1b8. Reverting to 1.1.0 gives this error:
```
[18/11/2021, 14:48:15] [SwitchBot] "The devices config section is missing the \"Device Type\" in the config, Check Your Conifg."
```

The 'Devices' tab in the switchbot plug settings says 'No Devices'.
and this is the config section:
```
{
            "name": "SwitchBot",
            "credentials": {
                "openToken": "XXXXXXXXXX",
                "notice": "Keep your token a secret!"
            },
            "options": {
                "devices": [
                    {
                        "bot": {
                            "deviceType": "outlet"
                        }
                    }
                ],
                "debug": "device"
            },
            "platform": "SwitchBot"
        }
```
 
When I remove the "bot": section from the config and restart I finally get access to my curtains again.

**Comment by lgranger** [comment]
Here's my config:
```
        {
            "name": "SwitchBot",
            "credentials": {
                "openToken": "<Token>",
                "notice": "Keep your token a secret!"
            },
            "platform": "SwitchBot"
        }
```

Here's also a gist with the switchbot portion of the home bridge logs: https://gist.github.com/lgranger/4a3f33bb3f694fd2150f34466fd9d06b

I've also tried the current beta and it's still not working. I've had to disable the plugin.

**Comment by donavanbecker** [comment]
@lgranger, can you try Beta: 

[![npm version](https://badgen.net/badge/homebridge-switchbot@beta/>=1.1.1-beta.10/orange)](https://www.npmjs.com/package/@switchbot/homebridge-switchbot)

I fixed the issue in beta.9 that was preventing the plugin from loading.

[Click here to find out how to install Beta](https://github.com/OpenWonderLabs/homebridge-switchbot/wiki/Beta-Version)

**Comment by donavanbecker** [comment]
Version 1.2.0 has been released, Please update me on this, and if it has been resolved with this version.

**Comment by lgranger** [comment]
Update fixed the issues. Thanks!
